sdk/core/System.BinaryData/src/BinaryData.cs

```diff
@@ -0,0 +1,260 @@
+                Bytes,
+                out ArraySegment<byte> data))
+            {
+                return s_encoding.GetString(data.Array, data.Offset, data.Count);
+            }
+            return s_encoding.GetString(Bytes.ToArray());
+        }
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a read-only stream.
+        /// </summary>
+        /// <returns>A stream representing the data.</returns>
+        public Stream ToStream() =>
+            new ReadOnlyMemoryStream(Bytes);
+
+        /// <summary>
+        /// Gets the <see cref="BinaryData"/> as <see cref="ReadOnlyMemory{Byte}"/>.
+        /// </summary>
+        /// <returns><see cref="BinaryData"/> as bytes.</returns>
+        public ReadOnlyMemory<byte> ToBytes() =>
+            Bytes;
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to the specified type using
+        /// <see cref="JsonSerializer"/>.
+        /// </summary>
+        /// <typeparam name="T">The type that the data should be
+        /// converted to.</typeparam>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to Json.</param>
+        /// <returns>The data converted to the specified type.</returns>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+        public T ToObjectFromJson<T>(JsonSerializerOptions? options = default)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+        {
+            using var memoryStream = new MemoryStream();
+            ToStream().CopyTo(memoryStream);
+            return (T) JsonSerializer.Deserialize(memoryStream.ToArray(), typeof(T), options);
+
+        }
+
+        /// <summary>
+        /// Implicit conversion to <see cref="ReadOnlyMemory{Byte}"/>.
+        /// </summary>
+        /// <param name="data"></param>
+        public static implicit operator ReadOnlyMemory<byte>(
+            BinaryData data) =>
+            data.Bytes;
+
+        /// <summary>
+        /// Implicit conversion to <see cref="ReadOnlySpan{Byte}"/>.
+        /// </summary>
+        /// <param name="data"></param>
+        public static implicit operator ReadOnlySpan<byte>(
+            BinaryData data) =>
+            data.Bytes.Span;
+
+        /// <summary>
+        /// Two <see cref="BinaryData"/> objects are equal if the memory regions point to the same array and have the
+        /// same length. The method does not check to see if the contents are equal.
+        /// </summary>
+        /// <param name="obj">The <see cref="BinaryData"/> to compare.</param>
+        /// <returns>true if the current instance and other are equal; otherwise, false.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object? obj)
+        {
+            if (obj is BinaryData data)
+            {
+                return data.Bytes.Equals(Bytes);
+            }
+            return false;
+        }
```

Comment:
I believe API review said that it should be standard reference equality (which is why it's EB-Never)

```suggestion
        /// <summary>
        /// Determines whether the specified object is equal to the current object.
        /// </summary>
        /// <param name="obj">The object to compare with the current object.</param>
        /// <returns>
        /// <see langword="true" /> if the specified object is equal to the current object; otherwise, <see langword="false" />.
        /// </returns>
        [EditorBrowsable(EditorBrowsableState.Never)]
        public override bool Equals(object? obj)
        {
            return ReferenceEquals(this, other);
        }
```